File: ts/tests/local/test-local.mjs
/**
 * Local package test — runs against the built dist via npm link.
 *
 * Prerequisites:
 *   1. cd sdk/ts && npm run build && npm link
 *   2. cd tests/local && npm link @imbrace/sdk
 *   3. Copy .env.example → tests/local/.env and fill in credentials
 *   4. node test-local.mjs
 */

import { readFileSync } from 'fs'
import { resolve, dirname } from 'path'
import { fileURLToPath } from 'url'

const __dir = dirname(fileURLToPath(import.meta.url))

// Manual .env loader (no dotenv dep required here)
try {
  const env = readFileSync(resolve(__dir, '.env'), 'utf8')
  for (const line of env.split('\n')) {
    const [k, ...v] = line.trim().split('=')
    if (k && !k.startsWith('#') && !(k in process.env)) {
      process.env[k] = v.join('=')
    }
  }
} catch {
  // .env optional — rely on shell env
}

import { ImbraceClient } from '@imbrace/sdk'

const API_KEY  = process.env.IMBRACE_API_KEY  ?? ''
const ORG_ID   = process.env.IMBRACE_ORGANIZATION_ID ?? ''
const BASE_URL = process.env.IMBRACE_GATEWAY_URL ?? 'https://app-gateway.dev.imbrace.co'

let passed = 0
let failed = 0

function ok(label) {
  console.log(`  ✓ ${label}`)
  passed++
}

function fail(label, err) {
  console.error(`  ✗ ${label}: ${err?.message ?? err}`)
  failed++
}

// ── 1. Instantiation ─────────────────────────────────────────────────────────

console.log('\n[1] Instantiation')
let client
try {
  client = new ImbraceClient({ apiKey: API_KEY || 'placeholder', gateway: BASE_URL, organizationId: ORG_ID })
  ok('ImbraceClient created')
} catch (e) {
  fail('ImbraceClient created', e)
  process.exit(1)
}

// ── 2. Resource surface ───────────────────────────────────────────────────────

console.log('\n[2] Resource surface')
const expected = [
  'auth', 'account', 'organizations', 'teams', 'settings',
  'channel', 'contacts', 'conversations', 'messages',
  'workflows', 'boards', 'ips', 'ai', 'marketplace',
  'agent', 'campaign', 'outbound', 'license', 'health',
  'sessions', 'schedule', 'platform',
]
for (const r of expected) {
  client[r] ? ok(`client.${r}`) : fail(`client.${r}`, 'undefined')
}

// ── 3. Live API calls (skipped if no key) ────────────────────────────────────

if (!API_KEY) {
  console.log('\n[3] Live API calls — SKIPPED (set IMBRACE_API_KEY in tests/local/.env)')
} else {
  console.log(`\n[3] Live API calls  (gateway: ${BASE_URL})`)

  const checks = [
    ['health.check()',                 () => client.health.check()],
    ['account.getAccount()',           () => client.account.getAccount()],
    ['channel.list()',                 () => client.channel.list()],
    ['contacts.list({ limit: 1 })',    () => client.contacts.list({ limit: 1 })],
    ['teams.list()',                   () => client.teams.list()],
    ['conversations.getViewsCount()',  () => client.conversations.getViewsCount()],
    ['boards.list()',                  () => client.boards.list()],
    ['agent.list()',                   () => client.agent.list()],
  ]

  for (const [label, fn] of checks) {
    try {
      const res = await fn()
      ok(`${label}  →  ${JSON.stringify(res).slice(0, 60)}`)
    } catch (e) {
      fail(label, e)
    }
  }
}

// ── Summary ───────────────────────────────────────────────────────────────────

console.log(`\n${'─'.repeat(50)}`)
console.log(`  ${passed} passed  |  ${failed} failed`)
if (failed > 0) process.exit(1)

File: py/tests/local/test_local.py
"""
Local package test — runs against the local source.
Usage:
    pip install -e .
    python tests/local/test_local.py
"""
import os
import warnings
import json
from imbrace import ImbraceClient

API_KEY      = os.environ.get("IMBRACE_API_KEY", "api_bf3be272-1bd1-4944-b167-5dd997f9302f")
ACCESS_TOKEN = os.environ.get("IMBRACE_ACCESS_TOKEN", "acc_a9ff3bfa-7a97-4b5e-9c02-a567d15755d4")
ORG_ID       = os.environ.get("IMBRACE_ORGANIZATION_ID", "org_ee7e1814-1a77-419b-b95b-871353d81657")
BASE_URL     = os.environ.get("IMBRACE_GATEWAY_URL", "https://app-gatewayv2.imbrace.co")

passed = 0
failed = 0

def ok(label):
    global passed
    print(f"  ✓ {label}")
    passed += 1

def fail(label, err):
    global failed
    print(f"  ✗ {label}: {err}")
    failed += 1

# [1] Instantiation
print("\n[1] Instantiation")
client = None
try:
    with warnings.catch_warnings():
        warnings.simplefilter("ignore")
        client = ImbraceClient(
            api_key=API_KEY or None,
            access_token=ACCESS_TOKEN or None,
            gateway=BASE_URL,
        )
    ok("ImbraceClient created")
except Exception as e:
    fail("ImbraceClient created", e)
    exit(1)

# [2] Resource surface
print("\n[2] Resource surface")
expected = [
    "auth",
    "account",
    "organizations",
    "teams",
    "settings",
    "channel",
    "contacts",
    "conversations",
    "messages",
    "workflows",
    "boards",
    "ips",
    "ai",
    "marketplace",
    "agent",
    "campaign",
    "outbound",
    "license",
    "health",
    "sessions",
    "schedule",
    "platform",
]
for r in expected:
    if hasattr(client, r):
        ok(f"client.{r}")
    else:
        fail(f"client.{r}", "undefined")

# [3] Live API calls
if not API_KEY and not ACCESS_TOKEN:
    print("\n[3] Live API calls — SKIPPED (set IMBRACE_API_KEY or IMBRACE_ACCESS_TOKEN)")
else:
    method = "API Key" if API_KEY else "Access Token"
    print(f"\n[3] Live API calls (gateway: {BASE_URL}, via: {method})")
    
    checks = [
        ("health.check()", lambda: client.health.check()),
        ("account.get()", lambda: client.account.get()),
        ("channel.list()", lambda: client.channel.list()),
        ("contacts.list(limit=1)", lambda: client.contacts.list(limit=1)),
        ("teams.list()", lambda: client.teams.list()),
        ("conversations.get_views_count()", lambda: client.conversations.get_views_count()),
        ("boards.list()", lambda: client.boards.list()),
        ("agent.list()", lambda: client.agent.list()),
    ]

    for label, fn in checks:
        try:
            res = fn()
            # If the response is a Pydantic model, convert to dict first
            output = res.model_dump() if hasattr(res, "model_dump") else res
            print(f"  ✓ {label}  →  {json.dumps(output)[:60]}")
            passed += 1
        except Exception as e:
            fail(label, e)

print(f"\n{'─' * 50}")
print(f"  {passed} passed  |  {failed} failed")
if failed > 0:
    exit(1)
